pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -276,6 +276,77 @@ def __call__(self, obj, *targs, **tkwargs):
     def __repr__(self):
         return f"<transform: {self._transform.__name__}>"
 
+    def __add__(self, other):
+        """Add two dispatchers or a dispatcher and a container to create a TransformProgram.
+
+        When adding dispatchers, they are converted to containers with no args or kwargs.
+
+        Args:
+            other: Another TransformDispatcher, TransformContainer, or TransformProgram to add.
+
+        Returns:
+            TransformProgram: A new program with this dispatcher followed by the other.
+        """
+        # Import here to avoid circular import
+        from .transform_program import TransformProgram  # pylint: disable=import-outside-toplevel
+
+        # Convert this dispatcher to a container (no args/kwargs)
+        self_container = TransformContainer(self)
+
+        if isinstance(other, TransformDispatcher):
+            other_container = TransformContainer(other)
+            # --- Fix: check if both are final transforms ---
+            if self_container.final_transform and other_container.final_transform:
+                raise TransformError(
+                    f"Both {self} and {other} are final transforms and cannot be combined."
+                )
+            return TransformProgram([self_container, other_container])
+        if isinstance(other, TransformContainer):
+            if self_container.final_transform and other.final_transform:
+                raise TransformError(
+                    f"Both {self} and {other} are final transforms and cannot be combined."
+                )
+            return TransformProgram([self_container, other])
+        if isinstance(other, TransformProgram):
+            program = TransformProgram([self_container])
+            return program + other
+        return NotImplemented
+
+    def __mul__(self, n):
+        """Multiply a dispatcher by an integer to create a program with repeated dispatchers.
+
+        Args:
+            n (int): Number of times to repeat this dispatcher.
+
+        Returns:
+            TransformProgram: A new program with this dispatcher repeated n times.
+        """
+        # Import here to avoid circular import
+        from .transform_program import TransformProgram  # pylint: disable=import-outside-toplevel
+
+        if isinstance(n, int):
+            if n < 0:
+                raise ValueError("Cannot multiply transform dispatcher by negative integer")
+            # Convert to container (no args/kwargs) and repeat
+            if self.final_transform and n > 1:
+                raise TransformError(
+                    f"{self} is a final transform and cannot be applied more than once."
+                )
+            container = TransformContainer(self)
+            return TransformProgram([container] * n)
+        return NotImplemented
```

Comment:
Personally I try to do what I can to minimize the level of nesting, but this is purely stylistic.

```suggestion
        if not isinstance(n, int):
            return NotImplemented

        if n < 0:
            raise ValueError("Cannot multiply transform dispatcher by negative integer")

        # Convert to container (no args/kwargs) and repeat
        if self.final_transform and n > 1:
            raise TransformError(
                f"{self} is a final transform and cannot be applied more than once."
            )

        container = TransformContainer(self)
        return TransformProgram([container] * n)
```